### ISSUE TITLE
Survival mode

### ISSUE THREAD

**Issue by ninja13151** [Willard21/MineKhan#25]
When will you add survival? I think it would be a great addition to the game!

**Comment by oblanmaN** [comment]
Survival would require monsters, which would require mobs, which would require AI and more advanced entities. A lot of work, but I'd imagine it's coming.

**Comment by Hacker1254** [comment]
This has been asked *3 time

NO, he has said he dosent have plans in the near future,

**Comment by CrimsonCodes0** [comment]
Actually, what about survival in peaceful mode just to start?

**Comment by GitHubium** [comment]
Yeah willard like just add survival smh can't possibly be that hard smh just program the 1.16 update smh plz add nether and bees too

**Comment by Hacker1254** [comment]
@GitHubium If its so easy, why do you add it? 
unless that was sarcasm, than im stupid

**Comment by SnehasishCodez** [comment]
Guys forget about survival willard should focus on peacefull mode since no mobs no AI and btw modder in khan academy already made a health system willard should use that and tweak with it a bit.

**Comment by SnehasishCodez** [comment]
i think his name is j.s coder in khan academy go to his spin-off of minekhan the project is called fall damage.

**Comment by Hacker1254** [comment]
> i think his name is j.s coder in khan academy go to his spin-off of minekhan the project is called fall damage.
 Can i get a Link?

**Comment by Hacker1254** [comment]
That Link Also Has Biomes

**Comment by Hacker1254** [comment]
So  i have a few plans to make a scuffed survival mode
at the moment, im upgrading the heart sys, i am also trying to figure out on how he added the bioems
im also gonna make it were when the "survival" Mode is on, you cant fly nor clip, but thats the easy part, having stuff drop and an empty inv, i wont be able to do
here is my github - https://github.com/Hacker1254/Buncker-Hill-Minekhan/tree/Murge
here is the khan one - https://www.khanacademy.org/computer-programming/minekhan/6206167318478848

**Comment by IcingHackz** [comment]
If survival mode was implemented, then this game would achieve ultra poggerss @Hacker1254  @GitHubium  @oblanmaN  @CrimsonCodes0  @SnehasishCodez

**Comment by theshinramen** [comment]
cringe

**Comment by IcingHackz** [comment]
lmfao what

**Comment by SnehasishCodez** [comment]
Willard should copy js. coders work but add some new teaks and maybe change the heart asset using a png or something or if he decides to just snatch the texture from Minecraft files

**Comment by SnehasishCodez** [comment]
ATTENTION PLEASE !!!
Check this project out as it's the closest to a full-on survival mode.
https://www.khanacademy.org/computer-programming/geekypeoplecraft/4805420559319040

it has biomes and the water is actually water. meaning it's hitbox free so the water its just like the one in Minecraft.
we could tweak around with the water to make lava, the mountain biome for snow peaked tops there are endless possibilities.
HOWEVER, my suggestion would be to use the health system from this project here:

https://www.khanacademy.org/computer-programming/fall-damage/5522736472080384
If all goes right this could be the beginning of minekhan survival beta 1.0

**Comment by Phi-001** [comment]
Wow great job for finding them. You can just go ahead and implement them yourself and [make a PR](https://www.youtube.com/watch?v=dSl_qnWO104) ask Willard to merge them.

**Comment by SnehasishCodez** [comment]
@Phi-001 Sure Thing

**Comment by Hacker1254** [comment]
> ATTENTION PLEASE !!!
Check this project out as it's the closest to a full-on survival mode.
https://www.khanacademy.org/computer-programming/geekypeoplecraft/4805420559319040

This link is dead

**Comment by Hacker1254** [comment]
> Willard should copy js. coders work but add some new teaks and maybe change the heart asset using a png or something or if he decides to just snatch the texture from Minecraft files

you can only use khan assets

**Comment by GuineaPigBridge** [comment]
Ye. the new one is this: https://www.khanacademy.org/computer-programming/new-webpage/5447935989858304

**Comment by Willard21** [comment]
@SnehasishCodez I mean, not to be rude, but I would never release something that low quality since I'm held to a much higher standard than anyone making spin-offs. If I half-implement something, it means I've made a mistake and 100 people are going to tell me about it. If other people half-implement something, it means they're making progress.

Fall damage is nice and would work in survival, but combining it with flying is so bad.

Making slightly transparent blue blocks that you can walk through would take like 2 minutes, but partially transparent textures aren't supported, so there's a 50/50 chance looking through them will just erase every block behind them, as can be seen in that spin-off.

Biomes are fine, but green wool as cactus is kinda cringe, and real cactus blocks are kinda hard to make since they're not cubes. A majority of biomes have some defining feature that's hard to correctly implement, which is why I've been putting it off.


Anyway, I'm not really interested in implementing survival myself. If the community can get a makeshift version working, then they're more than welcome to. But I told people I'd delay the survival update by a month every time anyone asked for it, and I'm a man of my word 😤

**Comment by GuineaPigBridge** [comment]
Wait, did anyone ask?

**Comment by GuineaPigBridge** [comment]
SHOOT!

**Comment by Hacker1254** [comment]
> @SnehasishCodez I mean, not to be rude, but I would never release something that low quality since I'm held to a much higher standard than anyone making spin-offs. If I half-implement something, it means I've made a mistake and 100 people are going to tell me about it. If other people half-implement something, it means they're making progress.
> 
> Fall damage is nice and would work in survival, but combining it with flying is so bad.
> 
> Making slightly transparent blue blocks that you can walk through would take like 2 minutes, but partially transparent textures aren't supported, so there's a 50/50 chance looking through them will just erase every block behind them, as can be seen in that spin-off.
> 
> Biomes are fine, but green wool as cactus is kinda cringe, and real cactus blocks are kinda hard to make since they're not cubes. A majority of biomes have some defining feature that's hard to correctly implement, which is why I've been putting it off.
> 
> Anyway, I'm not really interested in implementing survival myself. If the community can get a makeshift version working, then they're more than welcome to. But I told people I'd delay the survival update by a month every time anyone asked for it, and I'm a man of my word 😤

Yes i like your understanding, we are like modding it and i wont want this stuff into the game untill its fully done, like the early minecraft pistons

**Comment by Hacker1254** [comment]
> ATTENTION PLEASE !!!
> Check this project out as it's the closest to a full-on survival mode.
> https://www.khanacademy.org/computer-programming/geekypeoplecraft/4805420559319040
> 
> it has biomes and the water is actually water. meaning it's hitbox free so the water its just like the one in Minecraft.
> we could tweak around with the water to make lava, the mountain biome for snow peaked tops there are endless possibilities.
> HOWEVER, my suggestion would be to use the health system from this project here:
> 
> https://www.khanacademy.org/computer-programming/fall-damage/5522736472080384
> If all goes right this could be the beginning of minekhan survival beta 1.0

now mine is ;]

**Comment by GuineaPigBridge** [comment]
Yay!

**Comment by GuineaPigBridge** [comment]
But u still have a lot to do!

**Comment by Hacker1254** [comment]
Yea, we have to add a sys that will turn off the hearts in creative and on in survival

**Comment by Hacker1254** [comment]
we also have to add a credits

**Comment by GuineaPigBridge** [comment]
Ye. but the water isn't transparent, the ui is everywhere and we have too many dungeons.

**Comment by Hacker1254** [comment]
i fixed the dungeons 
i knew that issue

**Comment by Hacker1254** [comment]
transparent water is kinda hard, as i have no idea how to add a block texters

**Comment by GuineaPigBridge** [comment]
I will tell you that tomorrow. Although I have school, I should still be able to do it.

**Comment by Hacker1254** [comment]
yea it what i have been doing

**Comment by SnehasishCodez** [comment]
ummm.........

**Comment by CorruptedKarma** [comment]
The project must be deleted now...

**Comment by Hacker1254** [comment]
?

**Comment by SnehasishCodez** [comment]
yah ???

**Comment by CorruptedKarma** [comment]
> When will you add survival? I think it would be a great addition to the game!

Why dont you add it?

**Comment by DiamondCoder1000** [comment]
How about this one: https://minekhan.thingmaker.repl.co/

**Comment by Willard21** [comment]
What about it?

**Comment by DiamondCoder1000** [comment]
It has survival.

**Comment by DiamondCoder1000** [comment]
Also tools

**Comment by ImaginaryNeon** [comment]
Survival is selectable, but clicking Create World brings you to the Minecraft website, which is hilarious, but is pointless for Chromebook users like me.

**Comment by sauron65** [comment]
@ImaginaryNeon 

> Survival is selectable, but clicking Create World brings you to the Minecraft website, which is hilarious, but is pointless for Chromebook users like me. (cuz we can't download it)

Minecraft: Java Edition can run on ChromeOS, though this requires enabling Linux which many be blocked, and [Mojang recommends against this](https://help.minecraft.net/hc/en-us/articles/4408901759117-Can-Minecraft-Run-on-Chromebooks-)

**Comment by ImaginaryNeon** [comment]
> Guys forget about survival willard should focus on peacefull mode since no mobs no AI and btw modder in khan academy already made a health system willard should use that and tweak with it a bit.

It's not that easym
